File: rockets/src/main/kotlin/com/spacex/launches/rockets/datasources/remote/RocketsApiServiceModule.java
package com.spacex.launches.rockets.datasources.remote;

import android.content.Context;
import androidx.annotation.NonNull;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class RocketsApiServiceModule {

  @Provides
  static RocketsApiService provideRocketsApiService(@NonNull final Context context) {
    return new RocketsApiService(context);
  }
}

File: rockets/src/main/kotlin/com/spacex/rockets/datasources/remote/RocketsApiServiceModule.kt
package com.spacex.rockets.datasources.remote

import android.content.Context
import dagger.Module
import dagger.Provides

@Module
internal object RocketsApiServiceModule {

    @JvmStatic
    @Provides
    fun provideRocketsApiService(context: Context): RocketsApiService = RocketsApiService(context)
}
